pasture-algorithms/src/convexhull.rs

```diff
@@ -0,0 +1,596 @@
+/// or an error if less than 3 linearily independent points were given in the input buffer.
+pub fn convex_hull_as_triangle_mesh<T: PointBuffer>(buffer: &T) -> Result<Vec<(usize, usize, usize)>, &'static str> {
+    let triangles = create_convex_hull(buffer);
+    if triangles.len() < 2 {
+        return Err("input buffer cointains too few linearly independent points");
+    }
+    let mut triangle_indices = Vec::new();
+    for tri in triangles {
+        triangle_indices.push((tri.a, tri.b, tri.c));
+    }
+    return Ok(triangle_indices);
+}
+
+/// Convex Hull generation as points
+/// Returns the convex hull as a vector that contains the indices the points forming a convex hull around all input points.
+pub fn convex_hull_as_points<T: PointBuffer>(buffer: &T) -> Vec<usize> {
+    let triangles = create_convex_hull(buffer);
+    let mut points = HashSet::new();
+    for tri in triangles {
+        points.insert(tri.a);
+        points.insert(tri.b);
+        points.insert(tri.c);
+    }
+    let point_indices: Vec<usize> = points.into_iter().collect();
+    return point_indices;
+}
+
+fn create_convex_hull<T: PointBuffer>(buffer: &T) -> Vec<Triangle> {
+    let mut triangles: Vec<Triangle> = Vec::new();
+    let position_attribute = match buffer
+        .point_layout()
+        .get_attribute_by_name(POSITION_3D.name())
+    {
+        Some(a) => a,
+        None => return triangles,
+    };
+
+    let mut pointid: usize = 0;
+    if position_attribute.datatype() == POSITION_3D.datatype() {
+        for point in buffer.iter_attribute::<Vector3<f64>>(&POSITION_3D) {
+            iteration(buffer, pointid, point, &mut triangles);
+            pointid += 1;
+        }
+    } else {
+        for point in buffer.iter_attribute_as::<Vector3<f64>>(&POSITION_3D) {
+            iteration(buffer, pointid, point, &mut triangles);
+            pointid += 1;
+        }
+    };
+    return triangles;
+}
+
+/// Performs a single iteration of the cunvex hull algorithm.
+/// `pointid`: current index within the buffer
+/// `point`: current point to be checked against the convex hull, possibly extends the convex hull
+/// `triangles`: the set of triangles forming the convex hull
+/// Each iteration receives a convex hull and checks it against the current point within the buffer.
+/// If the point lies within the current convex hull no changes have to be made.
+/// If the point lies outside of the current convex hull the hull has to be extended to include the current point.
+/// If 'triangles' contain only one entry: no full triangle has been found yet. In case of linearily dependant points no second triangle is added.
+/// If all 'triangles' are in a plane with 'point' a special triangulation procedure is needed to prevent a degenerated triangle mesh.
+///
+/// TODO: compare to epsilons instead of 0.0
```

Comment:
I just removed this comment for now. Since mostly `f64` is used floating point errors shouldn't be a big concern for now as long as sensible value ranges are used. So far in my tests I didn't encounter a case where precision was an issue yet.